Add new seeders: eu.digibyteseed.com, testnet.digibyteseed.com, seed.digibyte.link, testnetseed.digibyte.link

saltedlolly opened the pull request:
- Remove loopback IPs by default. Instructions now explain how to add them if needed when running a full node on the same system.
- Improvements to documentation including reminding user to unblock port 53 on an external firewall.
- Remove testnet full node IP addresses. These were temporarily added since we had no working Seeders for testnet until recently.

The new seeders list is [here](https://onedrive.live.com/view.aspx?resid=7280E5EDD22CC6C4!4870&ithint=file%2cdocx&authkey=!AJ1dKLXo0Q9pd6w). More DigiByte Seeders are on their way.

We are trying to add seeders distributed around the world. US and EU now have two each. It would be great to get some added in Asia, Africa and Australia/NZ so we have broad global coverage. There is hopefully one being set up in Australia at the moment.

saltedlolly committed:
Add new mainnet seeder at nl.digibyteseed.com

- Remove loopback IPs by default. Instructions now explain how to add them if needed when ru nning a full node on the same system.
- Improvements to documentation including reminding user to unblock port 53 on an external firewall.

saltedlolly committed:
Update README.md

saltedlolly committed:
Update README.md

saltedlolly committed:
Remove sudo when running dnsseed - not needed

saltedlolly committed:
Add separate instructions for Ubuntu and Debian

saltedlolly committed:
Update README.md

saltedlolly committed:
Update README.md

saltedlolly committed:
Update README.md

saltedlolly committed:
Update README.md

saltedlolly committed:
Update README.md

saltedlolly committed:
Update README.md

saltedlolly committed:
Add seedstart.sh script template created by @ipwebnl

saltedlolly committed:
Improve startup script to make it easier to use

saltedlolly committed:
Update seedstartup.sh

saltedlolly committed:
Update README.md

saltedlolly committed:
Update README.md

saltedlolly committed:
Last one! A few minor tweaks to documentation

saltedlolly committed:
Update README.md

gto90 reviewed (approved):
ACK

Great work on this @saltedlolly !  Thank you very much for your contributions!

j50ng reviewed (approved):
ACK- good work

ycagel reviewed (approved):
aCK - Excellent work Olly!

saltedlolly committed:
Add new seeders URLs

New mainnet seeder: seed.digibytefoundation.io

New testnet seeders: testnet.digibyteseed.com, testnetseed.digibytefoundation.io

testnet.digibyteseed.com has been setup by @ipwebnl. It is currently located in Helsinki, Finland.

seed.digibytefoundation.io and testnetseed.digibytefoundation.io will be maintained by the DigiByte Foundation. These are not working yet but will be up in the next few days - I am working with Rudy and Rutger to get these setup. To broaden our global coverage they will be hosted in Singapore.

saltedlolly committed:
Add note that some providers do not allow you to add an NS record

saltedlolly committed:
Add note about choosing a good geographic location for your seeder

saltedlolly commented:
<img width="1603" alt="Screenshot 2023-02-18 at 11 20 33 am" src="https://user-images.githubusercontent.com/642592/219857577-fb596f57-54b3-491f-b330-75dcbfc83dbe.png">

I have been curious about the IP address hardcode on line 432. Not being a C++ developer I'm not certain what this does, but I am assuming it is a fallback IP address to a full node that the seeder can connect to in the event of a DNS connection problem. I did query the IP address to see if there was a full node there, but if there was it does not seem to be working. Can anyone explain what this is for? Am I correct in my assumption?

saltedlolly commented:
Secondly, at the moment when people want to run a testnet seeder, we are having them manually change line 25 in protocol.cpp from ```unsigned char pchMessageStart[4] = { 0xfa, 0xc3, 0xb6, 0xda };``` to ```unsigned char pchMessageStart[4] = { 0xfd, 0xc8, 0xbd, 0xdd };```.

I discovered in main.cpp, line 481 that there seems to already be an option for this:

```
  if (opts.fUseTestNet) {
      printf("Using testnet.\n");
      pchMessageStart[0] = 0xfd;
      pchMessageStart[1] = 0xc8;
      pchMessageStart[2] = 0xbd;
      pchMessageStart[3] = 0xdd;
      seeds = testnet_seeds;
      fTestNet = true;
  }
  ```
  
These seem to be the correct testnet magic bytes. This presumably means that if the user uses the --testnet flag at launch, then manually editing protocol.cpp is unnecessary to run a testnet seeder.  Can someone confirm?

saltedlolly committed:
Tweak instruction text

saltedlolly committed:
Rename start script to startseeder.sh

saltedlolly committed:
Add note that each person should only run one mainnet seeder

This helps to mitigate single points of failure, and encourages more participation.

saltedlolly committed:
Remove digibytefoundation.io seeders

Rutger says he doesn't have the technical skills to maintain their seeders, and we want each maintainer to take personal responsibility for their Seeder. We will find more volunteers to run them.

saltedlolly committed:
Add Seeders: seed.digibyte.link and testnetseed.digibyte.link

Also added a not requesting that each person run a maximum of one mainnet seeder and one testnet seeder each. This is to:
- Improve redundancy, by reducing the reliance on any one individual
- Improving decentralization
- Increase community participation by allowing more people to run a Seeder

saltedlolly commented:
After discussion with @j50ng, I have agreed to no longer add the new seeders myself. We will ask each maintainer to add their own seeder, via their own PR. This goes with the idea of everyone taking personal responsibility for their Seeder. I won't be adding any more seeders in this PR so you can finish approving it.

saltedlolly committed:
Fix small typo in iptables-persistent command
